File: STUDY/8. 정수론/1456. 거의 소수/1456.py
MAX=int(1e7)+1
a,b=map(int,input().split())
r=0
arr=[0]*MAX
for i in range(2,int(b**0.5)+1):
    if arr[i]: continue
    for j in range(i*2,MAX,i):
        arr[j]=1
for i in range(2,int(b**0.5)+1):
    if not arr[i]:
        j=i
        while i*j<=b:
            i*=j
            if a<=i<=b:
                r+=1
print(r)

File: STUDY/8. 정수론/1456. 거의 소수/1456.cpp
#include <iostream>
#include <cmath>
#include <vector>

using namespace std;

long long A, B;
int R;
const int MAX = int(1e7) + 1;
int arr[MAX];

int main() {
	cin >> A >> B;
	R = 0;
	for (int i = 2; i < int(sqrt(B)) + 1; i++) {
		if (arr[i] != 0) continue;
		for (int j = i * 2; j < MAX; j += i)
			arr[j] = 1;
	}
	for (int i = 2; i < int(sqrt(B)) + 1; i++) {
		if (arr[i] != 0) continue;
		long long j = i;
		int k = i;
		while (j <= B / k) {
			j *= k;
			if ((A <= j) && (B >= j)) {
				R++;
			}
		}
	}
	cout << R << "\n";
	return 0;
}